### PR TITLE
Fix inner callback call for non comparable items

### PR DESCRIPTION
**Pull request by Sumolari**

### CODE REVIEW

File: src/mergeForEach.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export type ComparisonResult = number
  * Can be used to implement efficient algorithms which profit from sorted data
  * like `mergeSort` or `mergeJoinWith`.
  *
- * Both collections must be sortable. They will be sorted ascendently using
+ * Both collections must be sortable. They will be sorted ascendantly using
```
Comment (furstenheim-geoblink) [inline src/mergeForEach.ts]: why the change? both are good

File: src/mergeForEach.ts [hunk 2]
```diff
@@ -61,13 +61,11 @@ function mergeForEach<
...
-      } else if (lhsValue > rhsValue) {
-        return SORTING_ORDER.LHS_AFTER_RHS
-      } else {
-        return SORTING_ORDER.EQUAL
-      }
+      if (lhsValue === rhsValue) return SORTING_ORDER.EQUAL
+      if (lhsValue < rhsValue) return SORTING_ORDER.LHS_BEFORE_RHS
+      if (lhsValue > rhsValue) return SORTING_ORDER.LHS_AFTER_RHS
+
+      throw new Error(`[@geoblink/lodash-mixins#mergeForEach] Found non-comparable values: ${lhsValue} on LHS and ${rhsValue} on RHS`)
```
Comment (furstenheim-geoblink) [inline src/mergeForEach.ts]: This might have impact where we use it. We have to be careful about this

File: src/mergeForEach.ts (hunk 1 above)
Comment (Sumolari) [reply src/mergeForEach.ts]: When I copied it to my comment in Github, Chrome's autocorrector complained about it 😆

File: src/mergeForEach.ts (hunk 2 above)
Comment (Sumolari) [reply src/mergeForEach.ts]: If we find this error then we were relying on the old bug which was actually an undefined/unexpected behaviour, I think it's better to fail even without us expecting it rather than giving false matches (this was happening in My Data wizard before and was a crazy thing to debug)

Comment (furstenheim-geoblink) [reply src/mergeForEach.ts]: I suppose
